pilot/pkg/features/pilot.go

```diff
@@ -757,6 +757,12 @@ var (
 	// This is a feature flag, can be removed if protobuf proves universally better.
 	KubernetesClientContentType = env.Register("ISTIO_KUBE_CLIENT_CONTENT_TYPE", "protobuf",
 		"The content type to use for Kubernetes clients. Defaults to protobuf. Valid options: [protobuf, json]").Get()
+
+	// This is an experimental feature flag, can be removed once it became stable, and should introduced to Telemetry API.
+	MetricRotationInterval = env.Register("METRIC_ROTATION_INTERVAL", 0*time.Second,
```

Comment:
In context of 'safe istio' and other discussions, I would like to remind that we have another pattern for experiments and features - labels and annotations on the Pods. 

There are good cases for global install-only unmodifiable options (env variables) - I don't think this is one of them, and probably we should start moving some of the other env variables. 

Could you define instead an annotation - maybe even without istio.io prefix since it doesn't seem specific to istio ? 
This would allow users to try out this feature on few canaries without changing the entire cluster.

Reply:
Sound good for an experimental feature.

Reply:
Long term a global default is probably the answer - that's what we do for opencensus (at 1hr). There's never a need to customize per workload IMHO.

Reply:
a draft implement in https://github.com/istio/istio/pull/44605/commits/f91fae2a905dc7d7f7b2257fc25674e6fd585ccd, cc @howardjohn @costinm